subsys/bluetooth/controller/Kconfig

```diff
@@ -433,28 +438,68 @@ endif # BT_CONN
 
 config BT_CTLR_CHAN_SEL_2
 	bool "Channel Selection Algorithm #2"
-	depends on (BT_CONN || BT_CTLR_ADV_EXT) && BT_CTLR_CHAN_SEL_2_SUPPORT
+	depends on (BT_CONN || BT_CTLR_ADV_PERIODIC) && BT_CTLR_CHAN_SEL_2_SUPPORT
 	default y
 	help
 	  Enable support for Bluetooth 5.0 LE Channel Selection Algorithm #2 in
 	  the Controller.
 
 config BT_CTLR_ADV_EXT
-	bool "LE Advertising Extensions"
+	bool "LE Advertising Extensions [EXPERIMENTAL]"
 	depends on BT_CTLR_ADV_EXT_SUPPORT
 	select BT_CTLR_SCAN_REQ_NOTIFY
-	select BT_CTLR_CHAN_SEL_2
+	default y if BT_EXT_ADV
 	help
 	  Enable support for Bluetooth 5.0 LE Advertising Extensions in the
 	  Controller.
 
-config BT_ADV_SET
+config BT_CTLR_ADV_SET
 	int "LE Advertising Extensions Sets"
 	depends on BT_CTLR_ADV_EXT
-	default 1
+	range 1 64
+	default BT_EXT_ADV_MAX_ADV_SET if BT_EXT_ADV
 	help
 	  Maximum supported advertising sets.
 
+config BT_CTLR_ADV_AUX_SET
+	int "LE Advertising Extensions Auxiliary Sets"
+	depends on BT_CTLR_ADV_EXT
+	range 1 BT_CTLR_ADV_SET if BT_CTLR_ADV_PERIODIC
+	range 0 BT_CTLR_ADV_SET
+	default 1 if BT_CTLR_ADV_PERIODIC
+	default BT_CTLR_ADV_SET
+	help
+	  Maximum supported advertising auxiliary channel sets.
+
+config BT_CTLR_ADV_PERIODIC
+	bool "LE Periodic Advertising"
+	depends on BT_CTLR_ADV_EXT
+	select BT_CTLR_CHAN_SEL_2
+	help
+	  Enable support for Bluetooth 5.0 LE Periodic Advertising in the
+	  Controller.
+
+config BT_CTLR_ADV_SYNC_SET
+	int "LE Periodic Advertising Sets"
+	depends on BT_CTLR_ADV_PERIODIC
+	range 1 BT_CTLR_ADV_AUX_SET
+	help
+	  Maximum supported periodic advertising sets.
+
+config BT_CTLR_ADV_DATA_LEN_MAX
+	int "Maximum Advertising Data Length"
+	depends on BT_CTLR_ADV_EXT
+	range 31 1650
+	help
+	  Maximum Extended Advertising Data Length.
+
+config BT_CTLR_SCAN_AUX_SET
```

Comment:
Could you give a brief explanation describing the differences between the following configurations:
* BT_CTLR_ADV_SET
* BT_CTLR_ADV_AUX_SET
* BT_CTLR_ADV_SYNC_SET
* BT_CTLR_SCAN_AUX_SET

I believe we need to make the descriptions more clear by making them match the specification better.
I would expect there to be only two configuration options:
1. BT_CTLR_ADV_SET. The maximum number of supported advertising sets. This covers legacy, extended, and periodic advertising sets
2. BT_CTLR_SYNC_SET. The maximum number of syncs to periodic advertisers.  

If this controller wants to expose more configuration options, I would propose to move them under the section for BT_LL_SW_SPLIT

Reply:
BT_CTLR_ADV_SET is the maximum number of supported advertising sets.
BT_CTLR_ADV_AUX_SET is maximum number of supported auxiliary channel sets, defines the number of actual required advertising instances on auxiliary channels. If users only need legacy advertising sets, they have the option to save a lot of memory and code space.
BT_CTLR_SCAN_AUX_SET is the maximum supported auxiliary channel scan sets, this configures maximum number of simultaneous extended advertising that is to be scanned/followed. A primary channel scan window could pick multiple extended advertisers, this Kconfig allows for how many to be scanned/followed.
BT_CTLR_ADV_SYNC_SET is equivalent to BT_CTLR_SYNC_SET.

Ok, I will move the non-HCI definable/exposable Kconfig into advanced features of BT_LL_SW_SPLIT.